contracts/ServiceManagerToken.sol

```diff
@@ -107,22 +114,56 @@ contract ServiceManagerToken is GenericManager {
     }
 
     /// @dev Updates a service in a CRUD way.
+    /// @param token ERC20 token address for the security deposit, or ETH.
     /// @param configHash IPFS hash pointing to the config metadata.
     /// @param agentIds Canonical agent Ids.
     /// @param agentParams Number of agent instances and required bond to register an instance in the service.
     /// @param threshold Threshold for a multisig composed by agents.
     /// @param serviceId Service Id to be updated.
     /// @return success True, if function executed successfully.
     function update(
+        address token,
         bytes32 configHash,
         uint32[] memory agentIds,
         IService.AgentParams[] memory agentParams,
         uint32 threshold,
         uint256 serviceId
     ) external returns (bool)
     {
-        return IService(serviceRegistry).update(msg.sender, configHash, agentIds, agentParams,
-            threshold, serviceId);
+        // Check for the zero address
+        if (token == address(0)) {
+            revert ZeroAddress();
+        }
+
+        if (token == ETH_TOKEN_ADDRESS) {
+            // Call the original ServiceRegistry contract function
+            IService(serviceRegistry).update(msg.sender, configHash, agentIds, agentParams, threshold, serviceId);
+            // Reset the service token-based data
+            // This function still needs to be called as the previous token could be a custom ERC20 token
+            IServiceTokenUtility(serviceRegistryTokenUtility).resetServiceToken(serviceId);
+        } else {
+            // Wrap agent params with just 1 WEI bond going to the original ServiceRegistry contract,
+            // and actual token bonds being recorded with the ServiceRegistryTokenUtility contract
+            uint256 numAgents = agentParams.length;
+            uint256[] memory bonds = new uint256[](numAgents);
+            for (uint256 i = 0; i < numAgents; ++i) {
+                // Copy actual bond values for each agent Id that has at least one slot in the updated service
+                if (agentParams[i].slots > 0) {
```

Comment:
Now, if agentParams[i].slots > 0, we also need to make sure that agentParams[i].bonds> 0

Reply:
added in the next PR